legal-api/src/legal_api/core/filing.py

```diff
@@ -90,16 +115,34 @@ def payment_account(self, account_identifier):
         self._payment_account = account_identifier
         self.storage.payment_account = self._payment_account
 
+    @property
+    def status(self) -> str:
+        """Return the status of this Filing."""
+        if not self._status and self._storage:
+            self._status = self._storage.status
+        return self._status
+
     @property
     def json(self) -> Optional[Dict]:
         """Return a dict representing the filing json."""
-        _json = {}
-        if self._storage:
-            if self._storage.status == Filing.Status.COMPLETED.value:
-                _json = VersionedBusinessDetailsService.get_revision(self.id, self._storage.business_id)
-            else:
-                return self.raw
-        return _json
+        if not self._storage or (self._storage and self._storage.status not in [Filing.Status.COMPLETED.value,
+                                                                                Filing.Status.PAID.value,
+                                                                                Filing.Status.PENDING.value,
+                                                                                ]):
+            return self.raw
+
+        filing_json = VersionedBusinessDetailsService.get_revision(self.id, self._storage.business_id)
```

Comment:
Versioned data is available only once the filing is complete because transaction id will be populated by the filer. For paid and pending shouldnt it add the diff to the raw json?